lib/puppet/provider/foreman_config_entry/cli.rb

```diff
@@ -52,9 +52,14 @@ def self.prefetch(resources)
     return if entries.nil?
     resources.each do |name, resource|
       provider = entries.find { |entry| entry.name == name }
-      if provider.nil? && resource[:ignore_missing]
-        # just consider it has a value we exepcted
-        provider = new(:name => name, :value => resource[:value])
+      if provider.nil?
+        # config_entry doesn't exist
+        if resource[:ignore_missing]
+          # just pretend it has the value we expect
+          provider = new(:name => name, :value => resource[:value])
+        else
+          next
+        end
       end
       resources[name].provider = provider
```

Comment:
It'd be more typical to add `if provider` at the end here, rather than change the flow of the loop above with `next`.